native/src/modules/navigation/InternalPathnameParser.js

```diff
@@ -0,0 +1,199 @@
+// @flow
+
+import {
+  CATEGORIES_ROUTE,
+  DASHBOARD_ROUTE,
+  DISCLAIMER_ROUTE,
+  EVENTS_ROUTE,
+  LANDING_ROUTE,
+  LOCAL_NEWS_TYPE,
+  NEWS_ROUTE,
+  OFFERS_ROUTE,
+  POIS_ROUTE,
+  SEARCH_ROUTE,
+  SPRUNGBRETT_OFFER_ROUTE,
+  TU_NEWS_TYPE
+} from 'api-client/src/routes'
+import type {
+  LocalNewsType,
+  TuNewsType
+} from 'api-client/src/routes'
+import type { RouteInformationType } from './createNavigate'
+
+const FEATURE_ID_INDEX = 3
+
+class InternalPathnameParser {
+  _pathname: string
+  _parts: Array<string>
+  _length: number
+  _languageCode: string
+  _fixedCity: string | null
+
+  constructor (pathname: string, languageCode: string, fixedCity: string | null) {
```

Comment:
Why is languageCode a part of the constructor? Shouldn't his be choosed automatically by parsing the pathname?

Reply:
Is this maybe some kind of fallback for routes which do not contain a language?

Reply:
The latter, yes. There are pathnames without a language in it. And I did not want to have to check afterwards for all routes if there is already a language included.

Reply:
Can you rename this to fallback? like this it is very unclear to me

Reply:
Will do in #135